### ISSUE TITLE
Cannot select gradient stop

### ISSUE THREAD

**Issue by openscript** [svgdotjs/svg.js#673]
Hello everybody

Upfront, something like a warning. It's the first time I'm using svg.js, so I'm not sure if it's really an issue.

I'm trying to update the color of a stop in a gradient. Unfortunately I struggling with selecting the stop.

I've created a [minimal example](https://jsfiddle.net/nez0t5t2/):
```
<svg width="100" height="100">
  <defs>
    <linearGradient id="gradient">
      <stop stop-opacity="0" offset="0"/>
      <stop stop-color="#000" offset="1"/>
    </linearGradient>
  </defs>
  <rect width="100" height="100" fill=url(#gradient)></rect>
</svg>
```
And now I try to update the color of the first stop to red:
`SVG.get('gradient').get(0).update(0, '#F00')`

Am I doing something wrong or is it a bug?

Robin

**Comment by Fuzzyma** [comment]
Hey Robin. Thanks for reporting.
`get` as well as `first` and `last()` return a node from the `childNode` array which _can_ be a textnode.
In your case it is the whitespaces between the tags.

We have to discuss if this is intended because mostly people only want elements.

So for now go with this quickfix

```js
SVG.get('gradient').children()[0].update(0, '#F00')
```

**Comment by openscript** [comment]
Hi Fuzzyma

Thank you for the quick response!

In my opinion it's okay to override the behaviour, because the intention is clear, that I want an SVG object, when I run `get` on an SVG.js object.

After reading the [documentation](http://svgjs.com/elements/#gradient-get) I came to the conclusion this should work, because the documentation states:

`var s1 = gradient.get(0) // -> returns "first" stop`

Robin

**Comment by Fuzzyma** [comment]
The major issue with this is performance because on every action we have to filter all elements. And this is big!
Furthermore the main use of svg.js is (or was) in building svg documents and querying them. Svg documents built by svg.js do not have whitespaces so we never came across this problem.

So I am thankful for this report

**Comment by Fuzzyma** [comment]
This is fixed in the 3.0 branch (but wont go into 2.x). We are using children instead of childNodes there so it should not happen.
Closing this here. Don't hesistate to post something here anyway
